--- a/RecCaloCommon/CMakeLists.txt
+++ b/RecCaloCommon/CMakeLists.txt
@@ -2,8 +2,6 @@
 # Package: RecCaloCommon
 ################################################################################
 
-find_package(boost_timer REQUIRED)
-find_package(boost_unit_test_framework REQUIRED)
 file(GLOB _sources src/*.cpp )
 
 gaudi_add_library(RecCaloCommon
@@ -28,6 +26,9 @@
 
 
 if(BUILD_TESTING)
+  find_package(boost_timer REQUIRED)
+  find_package(boost_unit_test_framework REQUIRED)
+
   gaudi_add_executable(IDMap_test.exe
     SOURCES tests/IDMap_test.cpp
     LINK DD4hep::DDCore Boost::timer
